sdk/tests/invariant.test.ts

```diff
@@ -0,0 +1,246 @@
+import type { Liquidity, Percentage } from 'invariant-cspr-wasm'
+import { ALICE, LOCAL_NODE_URL } from '../src/consts'
+import { Key, Network } from '../src/enums'
+import { Erc20 } from '../src/erc20'
+import { Invariant } from '../src/invariant'
+import { assertThrowsAsync, deployInvariantAndTokens, loadChai } from '../src/testUtils'
+import { callWasm, initCasperClient, loadWasm } from '../src/utils'
+
+let hashes: {
+  invariant: { loadHash: string; packageHash: string }
+  tokenX: { loadHash: string; packageHash: string }
+  tokenY: { loadHash: string; packageHash: string }
+}
+
+describe('invariant test', () => {
+  const client = initCasperClient(LOCAL_NODE_URL)
+  const deployer = ALICE
+  const network = Network.Local
+
+  beforeEach(async () => {
+    hashes = await deployInvariantAndTokens(client, deployer)
+  })
+
+  it('should change protocol fee', async () => {
+    const chai = await loadChai()
+    const invariant = await Invariant.load(client, hashes.invariant.loadHash, network)
+    const newFee: Percentage = { v: 20000000000n }
+    await invariant.changeProtocolFee(deployer, newFee)
+
+    const { protocolFee } = await invariant.getInvariantConfig()
```

Comment:
`protocolFee` is `Decimals` type here. You can easily add support for specific decimal types by including the type inside the parser. For example, in `decodeInvariantConfig`
```Typescript
const [protocolFee, remainder]: [Percentage, Uint8Array] = ...
```
please also include these changes into other parsers